test/pkg/compose/templatingdata/assembler.go

```diff
@@ -0,0 +1,392 @@
+
+	r, err := o.getReleaseFromRHOCP(minor)
+	if err != nil && !errors.Is(err, errNoRemoteRelease) {
+		return Release{}, err
+	}
+	if err == nil {
+		klog.InfoS("Found release in RHOCP repository", "minor", minor, "release", r)
+		return r, nil
+	}
+
+	r, err = o.getReleaseFromTheMirror(minor, false)
+	if err != nil && !errors.Is(err, errNoRemoteRelease) {
+		return Release{}, err
+	}
+	if err == nil {
+		klog.InfoS("Found release in RC mirror", "minor", minor, "release", r)
+		return r, nil
+	}
+
+	r, err = o.getReleaseFromTheMirror(minor, true)
+	if err != nil && !errors.Is(err, errNoRemoteRelease) {
+		return Release{}, err
+	}
+	if err == nil {
+		klog.InfoS("Found release in EC mirror", "minor", minor, "release", r)
+		return r, nil
+	}
+
+	klog.InfoS("No RPMs for the minor version found", "minor", minor)
+
+	return Release{}, errNoRemoteRelease
+}
+
+// getReleaseFromTheMirror looks for MicroShift RPM in OpenShift mirror
+func (o *TemplatingDataOpts) getReleaseFromTheMirror(minor int, devPreview bool) (Release, error) {
+	dp := ""
+	if devPreview {
+		dp = "-dev-preview"
+	}
+
+	repo := fmt.Sprintf("https://mirror.openshift.com/pub/openshift-v4/%s/microshift/ocp%s/latest-4.%d/el9/os/", getArch(), dp, minor)
+
+	resp, err := http.Get(repo)
+	if err != nil {
+		return Release{}, fmt.Errorf("http.Get(%s) failed: %w", repo, err)
+	}
+	if resp.StatusCode != 200 { // TODO: Maybe this should compare to 404?
+		return Release{}, errNoRemoteRelease
+	}
+	version, _, err := testutil.RunCommand(
+		"sudo", "dnf", "repoquery", "microshift", "--quiet",
+		"--queryformat", "%{version}-%{release}",
+		"--disablerepo", "*",
+		"--repofrompath", fmt.Sprintf("this,%s", repo),
+	)
+	if err != nil {
+		return Release{}, fmt.Errorf("failed to repoquery %q for microshift RPM: %w", repo, err)
+	}
+	relInfo, err := downloadReleaseInfoRPM(version, "--repofrompath", fmt.Sprintf("this,%s", repo))
+	if err != nil {
+		return Release{}, err
+	}
+
+	var images []string
+	if o.SkipContainerImagesExtraction {
+		klog.InfoS("Skipping container image extraction", "version", version)
+	} else {
+		images, err = getContainerImages(relInfo)
+		if err != nil {
+			return Release{}, err
+		}
+	}
+
+	return Release{
+		Repository: repo,
+		Version:    version,
+		Minor:      minor,
+		Images:     images,
+	}, nil
+}
+
+// getReleaseFromRHOCP looks for MicroShift RPM in RHOCP
+func (o *TemplatingDataOpts) getReleaseFromRHOCP(minor int) (Release, error) {
+	rhocp := fmt.Sprintf("rhocp-4.%d-for-rhel-9-%s-rpms", minor, getArch())
+	version, serr, err := testutil.RunCommand("sudo", "dnf", "repoquery", "microshift",
+		"--quiet",
+		"--queryformat", "%{version}-%{release}",
+		"--repo", rhocp,
+		"--latest-limit", "1",
+	)
+	if err == nil {
+		relInfo, err := downloadReleaseInfoRPM(version, "--repo", rhocp)
+		if err != nil {
+			return Release{}, err
+		}
+		var images []string
+		if o.SkipContainerImagesExtraction {
```

Comment:
Maybe check this before downloading the RPM?

Reply:
Ha, true :) thx